System headers missing when using `meson` (`dav1d`)

When using `meson`, like for `dav1d`, system headers appear to be missing.

For example, after running

```sh
git clone https://code.videolan.org/videolan/dav1d.git
cd dav1d
mkdir build
cd build
meson setup ..
```

I get this output:

```shell
> ../ia2/build/tools/rewriter/ia2-rewriter --output-prefix call_gates --root-directory ./ --output-directory ia2/ -p build src/data.c
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from ../src/data.c:32:
/usr/include/stdlib.h:32:10: fatal error: 'stddef.h' file not found
   32 | #include <stddef.h>
      |          ^~~~~~~~~~
looking at commands for src/data.c
No compilation command found for src/data.c
```

Or this:

```shell
> ../ia2/build/tools/rewriter/ia2-rewriter --output-prefix call_gates --root-directory ./ --output-directory ia2/ -p build src/decode.c
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
In file included from ../src/decode.c:31:
/usr/include/limits.h:124:16: fatal error: 'limits.h' file not found
  124 | # include_next <limits.h>
      |                ^~~~~~~~~~
looking at commands for src/decode.c
No compilation command found for src/decode.c
```

An example of one of the compile commands that is causing this:

```json
{
    "command": "cc -Isrc/libdav1d.so.7.0.0.p -Isrc -I../src -I. -I.. -Iinclude/dav1d -I../include/dav1d -Iinclude -I../include -I/home/kkysen/work/rust/dav1d/../ia2/runtime/libia2/include -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c99 -O3 -D_GNU_SOURCE -fvisibility=hidden -Wundef -Werror=vla -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-unused-parameter -Wstrict-prototypes -Werror=missing-prototypes -fomit-frame-pointer -ffast-math -DIA2_ENABLE=1 -std=gnu99 -Wno-strict-prototypes -Wno-missing-prototypes -fdiagnostics-color=always -fPIC -pthread -DPKEY=2 -MD -MQ src/libdav1d.so.7.0.0.p/lib.c.o -MF src/libdav1d.so.7.0.0.p/lib.c.o.d -o src/libdav1d.so.7.0.0.p/lib.c.o -c ../src/lib.c",
    "file": "../src/lib.c",
    "output": "src/libdav1d.so.7.0.0.p/lib.c.o"
}
```

--- a/docs/usage.md
+++ b/docs/usage.md
@@ -26,6 +26,22 @@ want to generate and use a `compile_commands.json` to ensure the rewriter
 preprocesses each source file with the same command-line arguments as when it is
 compiled.
 
+### Compile Command Paths
+
+Note that `libclangTooling`, which `ia2-rewriter` uses, is very finicky
+about relative paths, so we suggest making all relative paths absolute.
+Furthermore, `ia2-rewriter` itself sometimes uses paths as keys,
+so we suggest making all paths canonical.  We have a script,
+[`canonicalize_compile_command_paths.py`](../tools/rewriter/canonicalize_compile_command_paths.py),
+that does this automatically (doing its best to detect paths embedded in args).
+`libclangTooling` also requires compilation databases to be named exactly
+`compile_commands.json`, so this script assumes that as well,
+and must be run in the directory of the `compile_commands.json`, like this:
+
+```sh
+(cd $build_directory && $ia2_dir/tools/rewriter/canonicalize_compile_command_paths.py)
+```
+
 ## Manual source changes
 
 ### Defining compartments

--- a/tools/rewriter/canonicalize_compile_command_paths.py
+++ b/tools/rewriter/canonicalize_compile_command_paths.py
@@ -11,6 +11,16 @@ import json
 from pathlib import Path
 
 def main():
+    """
+    Assuming we're in a build directory containing a `compile_commands.json`,
+    canonicalize (including making absolute) all paths in every compile command,
+    including as part of arguments (detect as a best effort).
+
+    libclangTooling is finicky with relative paths,
+    and it and other tools don't always handle non-canonical paths well,
+    so making all paths in a `compile_commands.json` canonical is helpful.
+    """
+
     cc_db = Path("compile_commands.json")
     cc_text = cc_db.read_text()
     cmds = json.loads(cc_text)

